### ISSUE TITLE
Type signature of actions created by slices

### ISSUE THREAD

**Issue by RobertMenke** [reduxjs/rtk-github-issues-example#7]
Hey, I'm curious why the type definitions for the action creators we get back from `createSlice` expects 2 parameters when the action creators are only ever called with a single parameter.

<img width="805" alt="Screen Shot 2019-11-19 at 10 14 48 PM" src="https://user-images.githubusercontent.com/13398229/69206272-39808280-0b1a-11ea-900f-2be2525f65d2.png">

I see why it works, but would it make sense to update the type definition to use the signature that you're using in the docs?

**Comment by markerikson** [comment]
That doesn't sound right.  Can you post the code that generated the `displayRepo` action creator?

**Comment by RobertMenke** [comment]
Hey, this is from cloning the repo and then opening it up in IntelliJ. After clicking into the function IntelliJ does seem to think that the type definition should match

```typescript
    displayRepo(state, action: PayloadAction<CurrentRepo>) {
      const { org, repo } = action.payload
      state.org = org
      state.repo = repo
    },
```

After looking into the types a bit more I believe this actually may be an IntelliJ bug.

```typescript
declare type CaseReducerActions<CaseReducers extends SliceCaseReducerDefinitions<any, any>> = {
    [Type in keyof CaseReducers]: IfIsCaseReducerWithPrepare<CaseReducers[Type], ActionCreatorWithPreparedPayload<PrepareActionForReducer<CaseReducers[Type]>>, IfIsReducerFunctionWithoutAction<CaseReducers[Type], ActionCreatorWithoutPayload, PayloadActionCreator<PayloadForReducer<CaseReducers[Type]>>>>;
};
```

^^ The above doesn't seem to have any reference to state as an expected parameter, although the types are quite hard to parse at first glance :).

**Comment by markerikson** [comment]
> The above doesn't seem to have any reference to state as an expected parameter, although the types are quite hard to parse at first glance :).

Hah, no kidding :)  I'm the maintainer, and _I_ barely understand them! :)  I'm grateful that there's some TS experts in the Redux community who have been able to put those together for us.

The action creators definitely do not have any references to the state, either as an argument or a related type.  They _do_ reuse the type of the `action` argument for the reducer.  For this example, since `action` is a `PayloadAction<CurrentRepo>`, the action creator knows that it should return `{type: "issuesDisplay/displayRepo, payload: CurrentRepo}`.  It also will have a signature of `(payload: CurrentRepo) => PayloadAction<CurrentRepo>`.

**Comment by markerikson** [comment]
FWIW, I just opened up that same project in WebStorm 2019.1, and the type appears correct there too:

![image](https://user-images.githubusercontent.com/1128784/69208476-a9ded200-0b21-11ea-9587-222f2940b61f.png)

**Comment by RobertMenke** [comment]
Hmm yes looks like a Jetbrains bug... sigh I don’t think they use LSP for typescript. Thank you for looking into this one! I may open an issue in their issue tracker and reference this issue.

**Comment by markerikson** [comment]
Yeah, I was just about to do that, actually :)

**Comment by markerikson** [comment]
Went ahead and filed https://youtrack.jetbrains.com/issue/WEB-42559 .

I'll go ahead and close the issue, since this doesn't appear to be a problem in RTK itself.  Thanks for reporting it, though!

**Comment by luokuning** [comment]
> Went ahead and filed https://youtrack.jetbrains.com/issue/WEB-42559 .
> 
> I'll go ahead and close the issue, since this doesn't appear to be a problem in RTK itself. Thanks for reporting it, though!

I just vote for the request on youtrack.
